sql-plugin/src/main/340+/scala/com/nvidia/spark/rapids/shims/ShuffleOriginUtil.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import scala.collection.immutable.HashSet
+
+import org.apache.spark.sql.execution.exchange.{ENSURE_REQUIREMENTS, REBALANCE_PARTITIONS_BY_COL, REBALANCE_PARTITIONS_BY_NONE, REPARTITION_BY_COL, REPARTITION_BY_NUM, ShuffleOrigin}
+
+object ShuffleOriginUtil {
+  private val knownOrigins: HashSet[ShuffleOrigin] = HashSet(ENSURE_REQUIREMENTS,
```

Comment:
```suggestion
  private val knownOrigins = Set(ENSURE_REQUIREMENTS,
```

Reply:
Cleaned this up, but Scala type interference wasn't completely working, so still need to specify type on `val`, so looks like `private val knownOrigins: Set[ShuffleOrigin]`

Reply:
it's because of the the signature for `def isSupported(...)`, I should have suggested  `private val knownOrigins = Set[ShuffleOrigin](...)`